File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        if (root == null) return null;
        Stack<TreeNode> stack = new();
        stack.Push(root);
        while (stack.Count > 0) {
            var node = stack.Pop();
            if (node.left == null && node.right == null) continue;
            var left = node.left;
            node.left = node.right;
            node.right = left;
            if (node.left != null) stack.Push(node.left);
            if (node.right != null) stack.Push(node.right);
        }
        return root;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        out = [0] * len(temperatures)
        for i in range(len(temperatures)):
            while stack:
                if stack[-1][0] < temperatures[i]:
                    num, j = stack.pop()
                    out[j] = i - j
                else:
                    break
            stack.append((temperatures[i], i))
        return out